File: exercise/ts/day24/tests/ToyBuilder.ts
import {faker} from '@faker-js/faker';
import {Toy} from "../src/domain/Toy";
import {rightValue} from "./fp.extensions";
import {Time} from "./Time";

export class ToyBuilder {
    private readonly stock: number;

    private constructor(stock: number) {
        this.stock = stock;
    }

    static toysInStock(stock: number = 1): ToyBuilder {
        return new ToyBuilder(stock);
    }

    build(): Toy {
        return rightValue(Toy.create(Time.provider, faker.lorem.word(), this.stock));
    }
}

File: solution/day24/tests/Delivery.Tests/ToyBuilder.cs
namespace D.Tests;

public class ToyBuilder
{
    private readonly int _stock;

    private ToyBuilder(int stock) => _stock = stock;

    public static ToyBuilder ToysInStock(int stock = 1) => new(stock);

    public A Build() => A.Create(Time.Provider, Faker.Lorem.GetFirstWord(), _stock).RightUnsafe();
}